src/commands/dev-promote-semver.yml

```diff
@@ -0,0 +1,72 @@
+description: >
+  Uses the CLI to promote a dev version of an orb to the registry. 
+  This command supports Semantic Versioning support. In order to promote
+  the commit subject must include the `[semver:FOO]` pattern.
+  i.e. 'Merge pull request #XX from some-branch [semver:minor]' 
+
+parameters:
+  dev-version:
+    description: "A fully-qualified reference to a *development* orb. This takes the form <namespace>/<orb-name>@dev:<label>"
+    type: string
+
+  token-variable:
+    description: >
+      Name of env var containing your token. Pass this as a raw string such
+      as ORB_PUBLISHING_TOKEN. Do not paste the actual token into your
+      configuration. If omitted it's assumed the CLI has already been setup
+      with a valid token.
+    type: env_var_name
+    default: CIRCLE_TOKEN
+
+
+  add-pr-comment:
+    description: >
+      CircleCI can comment on the merged PR with the final production version. 
+      If you want contributors to see a message with the production version, 
+      please provide the username & token, and ensure the merge commit subject
+      includes the PR number in default format.
+      i.e. 'Merge pull request #XX from some-branch [semver:patch]' 
+    type: boolean
+
+  bot-user:
+    description: >
+      The GitHub user to post PR comment.
+    type: string
+    default: ""
+
+  bot-token:
+    description: >
+      The GitHub user's token to post PR comment.
+    type: string
+    default: ""
+
+steps:
+      - run:
+          name: Grab Commit Info
+          command: |
+            curl "https://circleci.com/api/v1.1/project/gh/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_BUILD_NUM}?circle-token=<< parameters.token-variable >>" > /tmp/build_info.txt            
+      - run:
+          name: Promote to prod
+          command: |
+            SEMVER_INCREMENT=`jq -r '.subject' /tmp/build_info.txt | sed -En 's/.*\[semver:(major|minor|patch)\].*/\1/p'`
```

Comment:
The commit message can be obtained from Git: `git log -1 --pretty=%s`.

No need for accessing the API, JSON, sed, etc. Though, you may still need the build_info file for the PR number later.

Reply:
Brilliant!  I think the command only uses the commit subject in both places, so git log is a nice change.